File: src/main/java/itmo/is/project/dto/user/spaceship/SpaceshipDto.java
package itmo.is.project.dto.user.spaceship;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import itmo.is.project.dto.user.UserDto;
import itmo.is.project.model.user.Spaceship;

public record SpaceshipDto(

        @Schema(example = "1")
        @JsonProperty(value = "id", required = true)
        Integer id,

        @JsonProperty(value = "pilot", required = true)
        UserDto pilot,

        @Schema(example = "S")
        @JsonProperty(value = "size", required = true)
        Spaceship.Size size
) {
}

File: src/main/kotlin/itmo/isproject/dto/user/spaceship/SpaceshipDto.kt
package itmo.isproject.dto.user.spaceship

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema
import itmo.isproject.dto.user.UserDto
import itmo.isproject.model.user.Spaceship

data class SpaceshipDto(

    @Schema(example = "1")
    @JsonProperty(value = "id", required = true)
    val id: Int?,

    @JsonProperty(value = "pilot", required = true)
    val pilot: UserDto,

    @Schema(example = "S")
    @JsonProperty(value = "size", required = true)
    val size: Spaceship.Size
)
